app/src/main/java/com/projectit3048c/MainViewModel.kt

```diff
@@ -57,10 +59,15 @@ class MainViewModel(var foodService : IFoodService =  FoodService()) : ViewModel
 
     fun fetchFoods() {
         viewModelScope.launch {
-           var innerFoods = foodService.fetchFoods()
-           foods.postValue(innerFoods!!)
+            try {
+                val innerFoods = foodService.fetchFoods() ?: throw Exception("Failed to fetch foods")
+                foods.postValue(innerFoods)
+            } catch (e: Exception) {
+                Log.e("MainViewModel", "Failed to fetch foods", e)
+            }
         }
```

Comment:
This is a bit of extra work, as it appears the only time an Exception would be thrown is on the if test above.

In lieu of a try-catch, what I recommend: move the log statement in the if test above.

Also, this only logs the exception, and doesn't do anything to heal the exception, so it's not necessarily handing the exception.  The program will continue to run after this, as if the exception never happened.